content/BUILD.gn

```diff
@@ -8,3 +8,17 @@ source_set("common") {
     "//ipc"
   ]
 }
+
+source_set("browser") {
+  sources = [
+    "browser/webui/brave_shared_resources_data_source.cc",
+    "browser/webui/brave_shared_resources_data_source.h",
+  ]
+
+  deps = [
+    "//base",
+    "//content/public/browser",
```

Comment:
you're missing `content/public/common` here for `content_features.h` and `url_constants.h`